src/routes/controllers/follow.js

```diff
@@ -0,0 +1,44 @@
+import models from '../../db/models';
+import { errorResponse, successResponse, checkUserByUsername } from '../utils/helpers';
+
+const { User } = models;
+
+/** ,
+ * Follow A User
+ * @param {object} req
+ * @param {object} res
+ * @returns {object} followed user object
+ */
+const follow = async (req, res) => {
+  const { url } = req;
+  const { userId } = req.params;
+  const { username } = req.params;
+
+  const user = await checkUserByUsername(req, res, username);
+  const follower = await User.findByPk(userId);
+  try {
+    if (url.includes('unfollow')) {
+      if (user.id === follower.id) {
+        return errorResponse(res, 409, 'You cannot unfollow yourself');
+      }
+      await user.removeFollowers(follower);
```

Comment:
I can't see where this function is implemented.

Reply:
They are called Sequelize magic methods, please permit me to explain further on monday during the TTL interaction so that I everybody can hear the explanations too.